File: packages/transform/src/features/http/CreateHttpParameterTransformer.ts
import { HttpParameterProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpParameterTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createParameter",
      write: HttpParameterProgrammer.write,
    });
}

File: packages/transform/src/features/http/CreateHttpParameterTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var CreateHttpParameterTransformer = createHttpParameterTransformerNamespace{}

type createHttpParameterTransformerNamespace struct{}

func (createHttpParameterTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpParameterProgrammer.Write(call.TypeText)
}
